### DISCUSSION TITLE
Support Rendering Mermaid Diagrams

### DISCUSSION THREAD

**Discussion by cpaika** [3rd/image.nvim#206] [Ideas]
It would be really cool to support rendering images from mermaid diagrams.  

So like you'd have a block of text:

```
graph LR
  hello --> goodbye
```

That would render when you weren't editing it into an image. (Similar to how Markdown works here in Github):

```mermaid
graph LR
  hello --> goodbye
```

This was mentioned here, unsure of where it landed: https://github.com/3rd/image.nvim/discussions/29#discussioncomment-7164850

**Comment by 3rd** [comment]
@cpaika
Thanks for the nudge! I want to keep image.nvim closer to its original purpose of being a library, and it's easy for things to get crazy if we let too many features slip in.
But I do love [diagrams](https://github.com/3rd/tsdiagram) and I had the day off, so I made a separate plugin for adding diagrams-as-code support: https://github.com/3rd/diagram.nvim
It should be easy to extend with both grammars and renderers, it has support just for markdown + mermaid now.

**Reply by 3rd** [reply]
One of my best speedruns :joy: 
![image](https://github.com/user-attachments/assets/2e538112-e125-4124-84c2-1b4117ac6992)

**Reply by cpaika** [reply]
LOL "I don't want to support this in this library for a good reason....but wrote it anyway".  

Very much appreciated :) excited to give a try today.  BTW whats that screenshot from? Looks like good task tracking software.

**Reply by 3rd** [reply]
Glad you're excited <3
Yep, that's my task tracker!
It lives [here](https://github.com/3rd/core/tree/master/core-cli) but it only works with my [fringe](https://github.com/3rd/syslang) document grammar. I have a lot of integrations and small tools that run on Syslang file repos ^_^

https://github.com/user-attachments/assets/0a5bce92-162f-4aa6-a585-00adf943920c
